sklearn/utils/tests/test_extmath.py

```diff
@@ -720,3 +721,52 @@ def test_safe_sparse_dot_dense_output(dense_output):
     if dense_output:
         expected = expected.toarray()
     assert_allclose_dense_sparse(actual, expected)
+
+
+class CustomLinearOperator(LinearOperator):
+    """
+        Since SciPy version 1.4.0 :func:`scipy.sparse.linalg.aslinearoperator`
+        should be used instead of this class. In earlier versions the
+        `LinearOperator` implementation do not have `_transpose()` method.
+    """
+    def __init__(self, M):
+        self.M = M
+        super(CustomLinearOperator, self).__init__(M.dtype, M.shape)
+
+    def _matvec(self, x):
+        return self.M @ x
+
+    def _transpose(self):
+        return CustomLinearOperator(self.M.T)
+
+
+def test_safe_sparse_dot_operator():
```

Comment:
This doesn't test the `ndim = 1` case it seems?

Reply:
LinearOperator is always 2D. Could you elaborate, please?

Reply:
> LinearOperator is always 2D

Didn't know that, nevermind then!